### ISSUE TITLE
`yarn docs:build` fails with `Error: error:0308010C:digital envelope routines::unsupported`

### ISSUE THREAD

**Issue by ivanlei** [Agoric/documentation#776]
**Summary**
Local builds and CI checks are failing `yarn docs:build` with `Error: error:0308010C:digital envelope routines::unsupported`

**Description**
See https://github.com/Agoric/documentation/pull/774 for an example of [failing checks](https://github.com/Agoric/documentation/actions/runs/4211520015/jobs/7309922099). The same errors reproduce in my local build.
```
➜  documentation git:(temp-new-content) yarn docs:build
yarn run v1.22.5
$ vuepress build main
wait Extracting site metadata...
tip Apply theme @vuepress/theme-default ...
tip Apply plugin container (i.e. "vuepress-plugin-container") ...
tip Apply plugin @vuepress/register-components (i.e. "@vuepress/plugin-register-components") ...
tip Apply plugin @vuepress/active-header-links (i.e. "@vuepress/plugin-active-header-links") ...
tip Apply plugin @vuepress/search (i.e. "@vuepress/plugin-search") ...
tip Apply plugin @vuepress/nprogress (i.e. "@vuepress/plugin-nprogress") ...
tip Apply plugin check-md (i.e. "vuepress-plugin-check-md") ...
tip Apply plugin @vuepress/last-updated (i.e. "@vuepress/plugin-last-updated") ...

● Client █████████████████████████ building (10%) 1/2 modules 1 active
 cache-loader › babel-loader › node_modules/@vuepress/core/lib/client/clientEntry.js

● Server █████████████████████████ building (10%) 1/2 modules 1 active
 cache-loader › babel-loader › node_modules/@vuepress/core/lib/client/serverEntry.js

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
/Users/ivanleichtling/source/documentation/node_modules/loader-runner/lib/LoaderRunner.js:114
			throw e;
			^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (/Users/ivanleichtling/source/documentation/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/Users/ivanleichtling/source/documentation/node_modules/webpack/lib/NormalModule.js:417:16)
    at handleParseError (/Users/ivanleichtling/source/documentation/node_modules/webpack/lib/NormalModule.js:471:10)
    at /Users/ivanleichtling/source/documentation/node_modules/webpack/lib/NormalModule.js:503:5
    at /Users/ivanleichtling/source/documentation/node_modules/webpack/lib/NormalModule.js:358:12
    at /Users/ivanleichtling/source/documentation/node_modules/loader-runner/lib/LoaderRunner.js:373:3
    at iterateNormalLoaders (/Users/ivanleichtling/source/documentation/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
    at iterateNormalLoaders (/Users/ivanleichtling/source/documentation/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
    at /Users/ivanleichtling/source/documentation/node_modules/loader-runner/lib/LoaderRunner.js:236:3
    at context.callback (/Users/ivanleichtling/source/documentation/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at /Users/ivanleichtling/source/documentation/node_modules/cache-loader/dist/index.js:134:7
    at /Users/ivanleichtling/source/documentation/node_modules/graceful-fs/graceful-fs.js:61:14
    at FSReqCallback.oncomplete (node:fs:197:23) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v18.12.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Impact**
Unable to make PRs or publish doc updates

**Environment**
OS: macOS 13.1 (22C65)
Processor: Apple M1
```
➜  documentation git:(temp-new-content) ✗ yarn --version
1.22.5
➜  documentation git:(temp-new-content) ✗ npm --version
8.19.2
➜  documentation git:(temp-new-content) ✗ node --version
v18.12.1
```

**Comment by ivanlei** [comment]
One potential workaround is setting `NODE_OPTIONS=--openssl-legacy-provider`. This does seem somewhat lame as it's a relaxing of TLS security.

```
➜  documentation git:(temp-new-content) NODE_OPTIONS=--openssl-legacy-provider yarn docs:build
yarn run v1.22.5
$ vuepress build main
wait Extracting site metadata...
tip Apply theme @vuepress/theme-default ...
tip Apply plugin container (i.e. "vuepress-plugin-container") ...
tip Apply plugin @vuepress/register-components (i.e. "@vuepress/plugin-register-components") ...
tip Apply plugin @vuepress/active-header-links (i.e. "@vuepress/plugin-active-header-links") ...
tip Apply plugin @vuepress/search (i.e. "@vuepress/plugin-search") ...
tip Apply plugin @vuepress/nprogress (i.e. "@vuepress/plugin-nprogress") ...
tip Apply plugin check-md (i.e. "vuepress-plugin-check-md") ...
tip Apply plugin @vuepress/last-updated (i.e. "@vuepress/plugin-last-updated") ...

✔ Client
  Compiled successfully in 7.19s

✔ Server
  Compiled successfully in 5.73s

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
wait Rendering static HTML...
success Generated static files in main/.vuepress/dist.

✨  Done in 9.48s.
```

**Comment by ivanlei** [comment]
Alternatively, it's supposedly possible to change the hash being used via webpack config. I messed with this a bit but wasn't successful. An example patch is below. I confirmed this patch alters what hash is being used by webpack but no valid value I chose seemed to work.
```
diff --git a/main/.vuepress/config.js b/main/.vuepress/config.js
index 0d8a1616..7089a9c3 100644
--- a/main/.vuepress/config.js
+++ b/main/.vuepress/config.js
@@ -71,6 +71,13 @@ module.exports = {
     ],
   ],

+  configureWebpack: {
+    output: {
+        hashFunction: "sha256",
+    },
+  },
+
   /* --- DEFAULT THEME CONFIG --- */
   themeConfig: {
     sidebarDepth: 1,
```

**Comment by gibson042** [comment]
This error stems from a hard-coded `createHash("md4")` that was [fixed in webpack v5.54.0](https://github.com/webpack/webpack/commit/4a8bf4c084f98bef8b47f0050c352f246ffb3527#diff-d34dd78b4ea6b37ac0551517562d61f39d0fc50699240a065ec8b13b37e2ed66) but has not been incorporated into VuePress, which is [stuck on v4](https://github.com/vuejs/vuepress/blob/5e3a5a098b5619f41c68b1e8daf6421d56194e77/packages/@vuepress/core/package.json#L68) and presumably will never be updated because **[VuePress is now in maintenance mode](https://github.com/vuejs/vuepress/tree/5e3a5a098b5619f41c68b1e8daf6421d56194e77#status)**. So a long-term fix will require replacing use of that, either with [VitePress](https://vitepress.vuejs.org/) as suggested or with something else.

**Comment by arirubinstein** [comment]
vuepress-next (2?) seems to be the successor, however it's still in beta.  I suspect once that is released, we can migrate to that.
